pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -440,12 +441,54 @@ func gracefulTerminationDurationFromConfig(operatorSpec *operatorv1.StaticPodOpe
 	return observedGracefulTerminationDuration, nil
 }
 
+func gogcFromConfig(config map[string]interface{}) (string, error) {
```

Comment:
The new config field consumed by this function is logically an int, but the runtime accepts both integers and the string "off" for the GOGC environment variable itself. The function signature is making the pedantic distinction (i.e. its return value holds a GOGC value), even though Go 1.18 only supports one non-int value that this function should never return. The precedent is there, so the idea was to avoid requiring the caller of gogcFromConfig to know what constitutes a valid GOGC value.

Do you have a strong preference against? I don't mind changing it, but I wrote it both ways and didn't feel there was a downside. It's future-proofed for the world in which we want gogcFromConfig to sometime return `100,flyingCars="yes, please"` without changing `manageTemplate`.